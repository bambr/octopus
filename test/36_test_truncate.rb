#!/usr/bin/ruby

$: << File.dirname($0) + '/lib'
require 'run_env'

class Env < RunEnv
  def config
    super :object_space => false
  end
end

Env.connect_eval do
  create_object_space 0, :index => {:type => :FASTTREE, :unique => 1, :field_0 => { :type => :STRING, :index => 0 , :sort_order => :DESC }}
  keys = []
  5.times  do |i|
    foo = "foo#{i}"
    insert [foo]
    keys << foo
  end
  select *keys
  3.times do truncate end
  select *keys
end
