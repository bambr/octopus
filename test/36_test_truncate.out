# box.create_object_space(0, {:index=>{:type=>:FASTTREE, :unique=>1, :field_0=>{:type=>:STRING, :index=>0, :sort_order=>:DESC}}})
""

# box.insert(["foo0"])
1

# box.insert(["foo1"])
1

# box.insert(["foo2"])
1

# box.insert(["foo3"])
1

# box.insert(["foo4"])
1

# box.select("foo0", "foo1", "foo2", "foo3", "foo4")
[["foo0"], ["foo1"], ["foo2"], ["foo3"], ["foo4"]]

# box.truncate()
""

# box.truncate()
""

# box.truncate()
""

# box.select("foo0", "foo1", "foo2", "foo3", "foo4")
[]

